### PR TITLE
[ios] Fixes wikipedia place description text in dark mode

### PR DESCRIPTION
**Pull request by ArtemX9**
Added CSS color property in PlacePageDescriptionViewController.swift that is being substituted by particular hexCode of color depending on whether night mode is active or not.

I have decided to retrieve colors from UIColor because in future theme accents might change and this would result in additional work for finding this controller and replacing colors in it.

Fixes: #5364, #3042
Dark theme:
Simulator Screenshot - iPhone 14 Pro - 2023-07-17 at 20 52 23
Light theme:
Simulator Screenshot - iPhone 14 Pro - 2023-07-17 at 20 52 49

### CODE REVIEW

File: iphone/Maps/Core/Theme/Extensions/UIColor+hexString.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
+extension UIColor {
+    var hexString: String {
+      let cgColorInRGB = cgColor.converted(to: CGColorSpace(name: CGColorSpace.sRGB)!, intent: .defaultIntent, options: nil)!
+              let colorRef = cgColorInRGB.components
```
Comment (biodranik) [inline iphone/Maps/Core/Theme/Extensions/UIColor+hexString.swift:4]: Code formatting looks wrong and does not conform to OM style.

File: iphone/Maps/Core/Theme/Extensions/UIColor+hexString.swift [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+              let r = colorRef?[0] ?? 0
+              let g = colorRef?[1] ?? 0
+              let b = ((colorRef?.count ?? 0) > 2 ? colorRef?[2] : g) ?? 0
```
Comment (biodranik) [inline iphone/Maps/Core/Theme/Extensions/UIColor+hexString.swift:7]: Is this step required after conversion to RGB color space above?

**Review by biodranik — COMMENTED** [review]
Is there an easier way to implement it?

File: iphone/Maps/Core/Theme/Extensions/UIColor+hexString.swift (hunk 1 above)
Comment (ArtemX9) [reply iphone/Maps/Core/Theme/Extensions/UIColor+hexString.swift]: fixed

File: iphone/Maps/Core/Theme/Extensions/UIColor+hexString.swift (hunk 2 above)
Comment (ArtemX9) [reply iphone/Maps/Core/Theme/Extensions/UIColor+hexString.swift]: sometimes the blue component is not presented, so we need to check for it and do fallback in this case

File: iphone/Maps/Core/Theme/Extensions/UIColor+hexString.swift (hunk 1 above)
Comment (ArtemX9) [reply iphone/Maps/Core/Theme/Extensions/UIColor+hexString.swift]: how about adding of https://github.com/realm/SwiftLint ?

**Review by biodranik — APPROVED** [review]
Thank you! Is there a more generic way to get appropriate color without an explicit if (darkmode) in the code?

**Comment by vng** [comment]
error: PlacePageDescriptionViewController.swift:7:237: error: value of type 'UIColor' has no member 'hexString'
?

**Comment by ArtemX9** [comment]
> Thank you! Is there a more generic way to get appropriate color without an explicit if (darkmode) in the code?

Since we are using WebView for rendering the content we should be able to provide the theme we currently using, since the user can select dark theme specifically for application I found this as the only way to render the page using specified font color. Another approach would be to parse the HTML and extract text and render it inside TextField

**Comment by ArtemX9** [comment]
> error: PlacePageDescriptionViewController.swift:7:237: error: value of type 'UIColor' has no member 'hexString'
> ?

hm, I loaded the branch on another PC and it looks like the file is not saved for the project 🤷 
i'll re-upload it

**Pull request merged**